testutils/clustermanager/e2e-tests/gke.go

```diff
@@ -247,8 +228,7 @@ func (gc *GKECluster) Acquire() error {
 				gc.operations.DeleteClusterAsync(gc.Project, region, request.Zone, clusterName)
 			}
 			// Retry another region if cluster creation failed.
-			// TODO(chaodaiG): catch specific errors as we know what the error look like for stockout etc.
-			if i != len(regions)-1 {
+			if i != len(regions)-1 && needsRetryCreation(err.Error()) {
 				errMsg = fmt.Sprintf("%sRetry another region %q for cluster creation", errMsg, regions[i+1])
```

Comment:
```suggestion
				errMsg = fmt.Sprintf("%s Retry another region %q for cluster creation", errMsg, regions[i+1])
```
Or we expect always space in the end?

Reply:
Yeah I think so. The msg is always ended with `\n` or space.